### PR TITLE
Add Relation functionality added to BE + Dummy FE Modal

### PR DESCRIPTION
**Pull request by Catpillers**

### CODE REVIEW

File: Relation-SPA/src/app/relation-table/relation-table.component.html [hunk 1]
```diff
@@ -2,20 +2,106 @@
...
+              <label for="city">City</label>
+              <input type="text" id="city" class="form-control" />
+            </div>
+            <div class="form-group">
+              <label for="street">Street</label>
+              <input type="text" id="street" class="form-control" />
+            </div>
+            <div class="form-group">
+              <label for="number">Street Number</label>
+              <input type="text" id="number" class="form-control" />
```
Comment (dmyhun-andersen) [inline Relation-SPA/src/app/relation-table/relation-table.component.html]: Is it stored in db as int? If so we can use type="number".

File: RelationInfrastructure/Repository/RelationRepository.cs [hunk 2]
```diff
@@ -30,5 +30,6 @@ public async Task<IEnumerable<Relation>> GetRelationList(Guid? categoryId)
                 .ThenInclude(_ => _.Country)
                 .Where(_ => _.RelationCategories.FirstOrDefault().CategoryId == categoryId).ToListAsync();
         }
+
```
Comment (dmyhun-andersen) [inline RelationInfrastructure/Repository/RelationRepository.cs]: remove

File: RelationsBll/Services/RelationService.cs [hunk 3]
```diff
@@ -24,8 +25,50 @@ public async Task<IEnumerable<Relation>> GetList(Guid? categoryId)
             return await _relations.GetRelationList(categoryId);
         }
 
-        public async Task AddRelation(Relation relation)
+        public async Task AddRelation(AddRelationModel relationModel)
         {
+            Relation relation = new Relation
```
Comment (dmyhun-andersen) [inline RelationsBll/Services/RelationService.cs]: use var

File: RelationsBll/Services/RelationService.cs [hunk 4]
```diff
@@ -24,8 +25,50 @@ public async Task<IEnumerable<Relation>> GetList(Guid? categoryId)
...
+            {
+                Name =  relationModel.Name,
+                FullName = relationModel.FullName,
+                EmailAddress = relationModel.EmailAddress,
+                TelephoneNumber = relationModel.TelephoneNumber,
+            };
+            
+            Country country = new Country
```
Comment (dmyhun-andersen) [inline RelationsBll/Services/RelationService.cs]: the same with var usage

File: RelationsBll/Services/RelationService.cs (hunk 3 above)
Comment (dmyhun-andersen) [reply RelationsBll/Services/RelationService.cs]: and can we move it to auto mapper?

File: Relation-SPA/src/app/relation-table/relation-table.component.html (hunk 1 above)
Comment (Catpillers) [reply Relation-SPA/src/app/relation-table/relation-table.component.html]: Corrected and now its properly typed as number;

File: RelationInfrastructure/Repository/RelationRepository.cs (hunk 2 above)
Comment (Catpillers) [reply RelationInfrastructure/Repository/RelationRepository.cs]: Indentation was removed;

File: RelationsBll/Services/RelationService.cs (hunk 3 above)
Comment (Catpillers) [reply RelationsBll/Services/RelationService.cs]: As i know the Auto-Mapper should stay simple and be spared of any "complex mapping" to different entities that will save important information to data-base; So i decide to do this "logic" by myself, so i can be sure about what is happening and required data goes to proper places;

File: RelationsBll/Services/RelationService.cs (hunk 4 above)
Comment (Catpillers) [reply RelationsBll/Services/RelationService.cs]: Corrected in all required places;

**Pull request merged**